lib/scripts.js

```diff
@@ -185,7 +180,9 @@ var scripts = {
       token
     ];
 
-    return execScript.apply(scripts, args).then(function(result){
+    return job.takeLock(null, !!job.lock)
+    .then(function(lock) { return lock ? execScript.apply(scripts, args) : -2; }, function(err) { if (!(err instanceof Redlock.LockError)) { throw err; } })
+    .then(function(result){
       switch (result){
```

Comment:
this could be formatted into several lines or refactored into a separate function